jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh

```diff
@@ -99,40 +233,93 @@ validate_keystore_legacy() {
+   # instead of adding the keystore configuration at JGroups protocol level
+   insert_elytron_tls_config_if_needed "${CONFIG_FILE}"
+ fi
+
+ local encrypt_conf_mode
+ getConfigurationMode "<!-- ##JGROUPS_ENCRYPT## -->" "encrypt_conf_mode"
+
+ local key_store_conf_mode
+ getConfigurationMode "<!-- ##ELYTRON_KEY_STORE## -->" "key_store_conf_mode"
+
+ local has_elytron_subsystem
+ local xpath="\"//*[local-name()='subsystem' and starts-with(namespace-uri(), 'urn:wildfly:elytron:')]\""
+ testXpathExpression "${xpath}" "has_elytron_subsystem"
+
  local jgroups_encrypt_protocol="${JGROUPS_ENCRYPT_PROTOCOL:=SYM_ENCRYPT}"
  local jgroups_encrypt=""
  local key_store=""
+
  case "${jgroups_encrypt_protocol}" in
   "SYM_ENCRYPT")
     log_info "Configuring JGroups cluster traffic encryption protocol to SYM_ENCRYPT."
     local jgroups_unencrypted_message="Detected <STATE> JGroups encryption configuration, the communication within the cluster WILL NOT be encrypted."
-    local keystore_warning_message=""
-    local has_elytron_tls_marker=$(has_elytron_tls "${CONFIG_FILE}")
-    local keystore_validation_state="";
-    if [ "${has_elytron_tls_marker}" = "true" ]; then
-        keystore_validation_state=$(validate_keystore "${JGROUPS_ENCRYPT_SECRET}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_KEYSTORE}")
-    else
-        keystore_validation_state=$(validate_keystore_legacy "${JGROUPS_ENCRYPT_SECRET}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_KEYSTORE_DIR}")
-    fi
+    local valid_state=""
 
-    if [ "${keystore_validation_state}" = "valid" ]; then
-        # first add the elytron key-store:
-        if [ "${has_elytron_tls_marker}" = "true" ]; then
-            key_store=$(create_elytron_keystore "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_KEYSTORE_TYPE}" "${JGROUPS_ENCRYPT_KEYSTORE_DIR}")
-            jgroups_encrypt=$(create_jgroups_elytron_encrypt_sym "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_ENTIRE_MESSAGE:-true}")
-        else
-            # compatibility with old marker, only used if new marker is not present
-            jgroups_encrypt=$(create_jgroups_elytron_legacy "${JGROUPS_ENCRYPT_KEYSTORE}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_KEYSTORE_DIR}")
-        fi
-    elif [ "${keystore_validation_state}" = "partial" ]; then
-        keystore_warning_message="${jgroups_unencrypted_message//<STATE>/partial}"
+    if [ "${has_elytron_subsystem}" -eq 0 ]; then
+      if [ "${key_store_conf_mode}" = "xml" ]; then
+        validate_keystore_and_create "xml"
+      elif [ "${key_store_conf_mode}" = "cli" ]; then
+        validate_keystore_and_create "cli"
+      elif [ "${CONFIG_ADJUSTMENT_MODE,,}" = "xml" ]; then
```

Comment:
There is a lot of context...What is the difference with"${ key_store_conf_mode}" = "xml". Is that is obvious, forget my question...

Reply:
This is here to cover the following case: User has strictly defined that he wants only replacement via xml markers, we have the Elytron subsystem but we do not have the Keystore marker, in that case, then replace the protocol using the "legacy" style, which means include the key store in the JGroups protocol itself.

Reply:
Not so obvious indeed

Reply:
Thank-you, perhaps a comment would help?

Reply:
Yes, done